controllers/auth_config_status_updater.go

```diff
@@ -41,41 +46,152 @@ func (u *AuthConfigStatusUpdater) Reconcile(ctx context.Context, req ctrl.Reques
 	} else {
 		// resource found and it is to be watched by this controller
 		// we need to update its status
-		if err := u.updateAuthConfigStatus(ctx, req.String(), &authConfig, true); err != nil {
-			logger.Info(err.Error())
+		if err := u.updateAuthConfigStatus(log.IntoContext(ctx, logger), req.String(), &authConfig); err != nil {
 			return ctrl.Result{Requeue: true}, nil
 		} else {
-			logger.Info("resource status updated")
 			return ctrl.Result{}, nil
 		}
 	}
 }
 
-func (u *AuthConfigStatusUpdater) updateAuthConfigStatus(ctx context.Context, cacheId string, authConfig *api.AuthConfig, ready bool) error {
-	if len(u.Cache.FindKeys(cacheId)) == 0 {
-		return fmt.Errorf("resource not ready")
+func (u *AuthConfigStatusUpdater) updateAuthConfigStatus(ctx context.Context, cacheId string, authConfig *api.AuthConfig) (err error) {
+	logger := log.FromContext(ctx)
+
+	var condition api.Condition
```

Comment:
nit: would it be reasonable to remove the else and just have var condition api.Condition = statusReady()